### PR TITLE
Add dacpac references to sqlproj with relative path

### LINKED ISSUE #14672: db project references should use relative path when dacpac inside project dir
- Azure Data Studio Version:
Version: 1.26.1 (user setup)
Commit: 796abbdf5fdd341e3528e30a6b777c93e115eb02
Date: 2021-02-24T23:42:55.085Z
VS Code: 1.48.0
Electron: 9.4.3
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Windows_NT x64 10.0.17763

Steps to Reproduce:
1. Project Viewlet > Add Database Reference > choose dacpac inside the the project dir

This causes all db references to be re-written with full paths in .sqlproj file as seen here:

The problem is these should be relative paths to avoid the following issues:

- prevents a diff in source control - other users might have different local repo dir
- uses old paths from some other source (for example, I changed them to relative paths in sqlproj but doing Step#1 rewrote them to the old path - not sure where it knows the old paths as I don't see anything in workspace file.
- is it necessary to write all refs? why not write just the ref I am adding - this would allow us to customize sqlproj to supress build warnings

Thanks for listening.

### PR DESCRIPTION
**Pull request by kisantia**
This PR fixes #14672. This saves dacpac references with their relative path instead of the full file path. Saving the relative path makes more sense because the project and its dacpac references could be in source control. Saving the full filepath in the sqlproj would cause it to not work on different machines since people could have their repo in different locations. 

error message if dacpac is on a different drive:
image

### CODE REVIEW

File: extensions/sql-database-projects/src/controllers/projectController.ts [hunk 1]
```diff
@@ -580,7 +580,10 @@ export class ProjectsController {
 			} else if ((<ISystemDatabaseReferenceSettings>settings).systemDb !== undefined) {
 				await project.addSystemDatabaseReference(<ISystemDatabaseReferenceSettings>settings);
 			} else {
-				await project.addDatabaseReference(<IDacpacReferenceSettings>settings);
+				// update dacpacFileLocation to relative path to project file
+				const dacpacRefSettings = settings as IDacpacReferenceSettings;
+				dacpacRefSettings.dacpacFileLocation = vscode.Uri.file(utils.trimUri(vscode.Uri.file(project.projectFilePath), dacpacRefSettings.dacpacFileLocation));
```
Comment (Benjin) [inline extensions/sql-database-projects/src/controllers/projectController.ts:585]: https://nodejs.org/api/path.html#path_path_relative_from_to

I think would be a better choice than `trimUri()`

**Review by Benjin — APPROVED** [review]

File: extensions/sql-database-projects/src/controllers/projectController.ts [hunk 2]
```diff
@@ -580,7 +580,10 @@ export class ProjectsController {
 			} else if ((<ISystemDatabaseReferenceSettings>settings).systemDb !== undefined) {
 				await project.addSystemDatabaseReference(<ISystemDatabaseReferenceSettings>settings);
 			} else {
-				await project.addDatabaseReference(<IDacpacReferenceSettings>settings);
+				// update dacpacFileLocation to relative path to project file
+				const dacpacRefSettings = settings as IDacpacReferenceSettings;
+				dacpacRefSettings.dacpacFileLocation = vscode.Uri.file(path.relative(project.projectFolderPath, dacpacRefSettings.dacpacFileLocation.fsPath));
```
Comment (udeeshagautam) [inline extensions/sql-database-projects/src/controllers/projectController.ts:585]: do we have a way to detect if its different drive for windows and give appropriate error there...

Comment (kisantia) [reply extensions/sql-database-projects/src/controllers/projectController.ts:585]: added screenshot of the error message to the PR description

**Review by udeeshagautam — APPROVED** [review]

File: extensions/sql-database-projects/src/controllers/projectController.ts (hunk 2 above)
Comment (kisantia) [reply extensions/sql-database-projects/src/controllers/projectController.ts:585]: for all. Are there any concerns for having it not be specifically for windows?

Comment (udeeshagautam) [reply extensions/sql-database-projects/src/controllers/projectController.ts:585]: no not really, was not sure if that's needed for others is all. didnt notice the issue with any other.

**Pull request merged**